data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -134,9 +134,13 @@ protected long pos(T record) {
 
       Iterable<CloseableIterable<Record>> deleteRecords = Iterables.transform(deletes,
           delete -> openDeletes(delete, deleteSchema));
-      StructLikeSet deleteSet = Deletes.toEqualitySet(
-          // copy the delete records because they will be held in a set
-          CloseableIterable.transform(CloseableIterable.concat(deleteRecords), Record::copy),
+
+      // copy the delete records because they will be held in a set
+      CloseableIterable<Record> records = CloseableIterable.transform(CloseableIterable.concat(deleteRecords),
+          Record::copy);
+
+      StructLikeSet deleteSet = Deletes.toEqualitySet(CloseableIterable.transform(records,
+          record -> new InternalRecordWrapper(deleteSchema.asStruct()).wrap(record)),
```

Comment:
We prefer line wrapping that keeps arguments to the same method aligned rather than aligning arguments to different levels. Here, the lambda to create an `InternalRecordWrapper` is an argument to `transform`, but it is aligned with `deleteSchema.asStruct()` that is an argument to the outer `toEqualitySet` call.

Instead, can you add a newline for each argument to `toEqualitySet`? If the line with the lambda is too long, then you can also add a newline for it that it indented from the start of the line with `transform`, so it is clear that it is an argument to `transform` and not `toEqualitySet`.